src/deluge/io/midi/midi_follow.cpp

```diff
@@ -766,29 +766,43 @@ void MidiFollow::readDefaultsFromFile() {
 /// compares param name tag to the list of params available are midi controllable
 /// if param is found, it loads the CC mapping info for that param into the view
 void MidiFollow::readDefaultMappingsFromFile() {
-	char const* paramName;
 	char const* tagName;
+	bool foundParam;
 	while (*(tagName = storageManager.readNextTagOrAttributeName())) {
+		foundParam = false;
 		for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
 			for (int32_t yDisplay = 0; yDisplay < kDisplayHeight; yDisplay++) {
-				if (!strcmp(tagName, params::paramNameForFile(params::Kind::PATCHED,
-				                                              patchedParamShortcuts[xDisplay][yDisplay]))) {
-					paramToCC[xDisplay][yDisplay] = storageManager.readTagOrAttributeValueInt();
-				}
-				else if (!strcmp(tagName,
-				                 params::paramNameForFile(
-				                     params::Kind::UNPATCHED_SOUND,
-				                     params::UNPATCHED_START + unpatchedNonGlobalParamShortcuts[xDisplay][yDisplay]))) {
-					paramToCC[xDisplay][yDisplay] = storageManager.readTagOrAttributeValueInt();
-				}
-				else if (!strcmp(tagName,
-				                 params::paramNameForFile(params::Kind::UNPATCHED_GLOBAL,
-				                                          params::UNPATCHED_START
-				                                              + unpatchedGlobalParamShortcuts[xDisplay][yDisplay]))) {
-					paramToCC[xDisplay][yDisplay] = storageManager.readTagOrAttributeValueInt();
+				//let's see if this x, y corresponds to a valid param shortcut
+				if ((patchedParamShortcuts[xDisplay][yDisplay] != kNoParamID)
+				    || (unpatchedNonGlobalParamShortcuts[xDisplay][yDisplay] != kNoParamID)
+				    || (unpatchedGlobalParamShortcuts[xDisplay][yDisplay] != kNoParamID)) {
+					//if we have a valid param shortcut, let's confirm the tag name corresponds to that shortcut
+					if ((!strcmp(tagName, params::paramNameForFile(params::Kind::PATCHED,
```

Comment:
We should check if the arrays do not have `kNoParamID` case by case. So something like 

`(patchedParamShortcuts[xDisplay][yDisplay] != kNoParamID && !strcmp(tagName, params::paramNameForFile(params::Kind::PATCHED, patachedParamShortcuts[xDisplay][yDisplay])) || unpatchedNonGlobal || unpatchedGlobal`